plugins/formal-verification/Uniqueness.md

```diff
@@ -0,0 +1,187 @@
+# Aliasing problem overview
+
+As described in [The Geneva Convention][1] aliasing between references
+can make it difficult to verify simple programs.
+Let's consider the following Hoare formula:
+
+_{x = true} y := false {x = true}_
+
+If _x_ and _y_ refers to the same
+reference (they are aliased) the formula is not valid. \
+Aliasing is not only a problem for formal verification, but also for the programmer resulting in mysterious bugs as
+variables change their values seemingly on their own.
+
+The set of (object address) values associated with variables during the execution of a method is a context. It is only
+meaningful to speak of aliasing occurring within some context; if two instance variables refer to a single object, but
+one of them belongs to an object that cannot be reached from anywhere in the system, then the
+aliasing is irrelevant.
+
+Within any method, objects may be accessed through paths rooted at any of:
+
+- Self
+- An anonymous locally constructed object
+- A method argument
+- A result returned by another method
+- A (global) variable accessible from the method scope
+- A local method variable bounded to any of the above
```

Comment:
If this is taken from somewhere, please add a reference and phrase it as such: e.g. "Alice and Bob write in [AB95] that ..."  This also helps clarify when we're making claims about Kotlin, and when these are general OO considerations.